mongodump/mongodump.go

```diff
@@ -162,9 +162,31 @@ func (dump *MongoDump) Init() error {
 	return nil
 }
 
+func (dump *MongoDump) verifyCollectionExists() (bool, error) {
+	// Running MongoDump against a DB with no collection specified works. In this case, return true so the process
+	// can continue.
+	if dump.ToolOptions.Namespace.Collection == "" {
+		return true, nil
+	}
+
+	coll := dump.SessionProvider.DB(dump.ToolOptions.Namespace.DB).Collection(dump.ToolOptions.Namespace.Collection)
+	collInfo, err := db.GetCollectionInfo(coll)
+	if err != nil {
+		return false, err
+	}
+
+	return collInfo != nil, nil
+}
+
 // Dump handles some final options checking and executes MongoDump.
 func (dump *MongoDump) Dump() (err error) {
 	defer dump.SessionProvider.Close()
+
+	exists, err := dump.verifyCollectionExists()
+	if err != nil || !exists {
+		return err
```

Comment:
I think this should be split into two cases: 
- `err != nil` -> return an error message like "couldn't verify collection info" or something
- !exists -> log that the collection doesn't exist, then return nil.

It could be argued that dumping a non-existent collection should be an error case, but I'm not sure if that's true or desired.

Reply:
I agree there's something to be decided here. I wasn't sure what the right behavior should be. mongoexport has a command line option for asserting that the collection exists and an error is thrown if it doesn't. For now, I'll update it to match your suggestion but I think this should be discussed more when the cmd opts rationalization project is done.